### PR TITLE
Pr/270/typing and more

### PR DESCRIPTION
**Pull request by E1337Kat**
Augments #270 with some type fixes and whatnot.

### CODE REVIEW

File: src/features.ts [hunk 1]
```diff
@@ -18,6 +18,9 @@ const boolAsFeature = (currentState: boolean): FeatureState =>
 export const IsFeatureEnabled = (featureState: FeatureState): boolean =>
   featureState === FeatureState.Enabled;
 
+export const IsDynamicFeatureEnabled = (featureState: Dynamic<FeatureState>): boolean =>
+  featureState === ((): FeatureState => FeatureState.Enabled);
```
Comment (effs) [inline src/features.ts:22]: I'm not sure if I'm missing something here, but I think this needs to be `featureState() === FeatureState.Enabled` to actually evaluate the setting at the time? And/or use `IsFeatureEnabled(features.REDmodAutoconvertArchives())` (I do sort of like the `IsDynamicFeatureEnabled` naming, it's more explicit than just a function call.)

The reasoning behind the types might be a little obtuse, I meant to leave them somewhat verbose to try to make sure we're always intentional about whether something is static or dynamic at the callsite, but it should be a matter of property access vs. function call.

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -42,6 +42,7 @@ jobs:
           pushd ..
           npm install node-gyp
           npm install autogypi
```
Comment (effs) [inline .github/workflows/testing.yml:44]: I think we should be able to drop the gyp for now, btw, I forgot to change this when I dropped the depss.

Comment (E1337Kat) [reply .github/workflows/testing.yml]: good point

File: src/features.ts (hunk 1 above)
Comment (E1337Kat) [reply src/features.ts]: ah ya, I realized that later lol

File: .github/workflows/testing.yml (hunk 2 above)
Comment (E1337Kat) [reply .github/workflows/testing.yml]: It can be removed later, maybe?

**Pull request merged**